### PR TITLE
cam6_4_014: Remove WACCMX gridded component layer

### LINKED ISSUE #1055: Remove the ESMF gridded component layer in WACCMX
### Issue Type

Code Clean-up

### Issue Description

With the use of ESMF NUOPC mediator component coupling in CESM, the ESMF gridded component to couple the ionosphere to CAM physics is no longer needed.  This was introduced by @gold2718 to enable multiple instances of WACCMX when MCT component coupling was used.  MCT coupling is no longer used and the ESMF gridded component coupling in WACCMX should be removed.

### Will this change answers?

No

### Will you be implementing this yourself?

Yes

### PR DESCRIPTION
**Pull request by fvitt**
This removes the ESMF gridded component layer in waccmx coupling.

Closes #1055

### CODE REVIEW

**Review by gold2718 — APPROVED** [review]
Nice cleanup!

File: src/ionosphere/waccmx/edyn_phys_grid.F90 [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    if (numOwnedElements /= total_cols) then
+       write(tempc1,'(i10)') numOwnedElements
+       write(tempc2,'(i10)') total_cols
+       call endrun(trim(subname)//": ERROR numOwnedElements "// &
+            trim(tempc1) //" not equal to local size "// trim(tempc2))
+    end if
+
+    allocate(ownedElemCoords(spatialDim*numOwnedElements))
+    allocate(lonMesh(total_cols), latMesh(total_cols))
```
Comment (cacraigucar) [inline src/ionosphere/waccmx/edyn_phys_grid.F90:94]: Need to put checks on all the allocations in this subroutine

**Review by cacraigucar — CHANGES_REQUESTED** [review]

File: src/ionosphere/waccmx/edyn_phys_grid.F90 (hunk 1 above)
Comment (fvitt) [reply src/ionosphere/waccmx/edyn_phys_grid.F90]: done

File: src/ionosphere/waccmx/edyn_phys_grid.F90 [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+       ncols = get_ncols_p(chnk)
+       do col = 1, ncols
+          dindex = dindex + 1
+          decomp(dindex) = get_gcol_p(chnk, col)
+       end do
+    end do
+
+    ! Create a DistGrid based on the physics decomp
+    dist_grid_2d = ESMF_DistGridCreate(arbSeqIndexList=decomp, rc=rc)
+    call edyn_esmf_chkerr(subname, 'ESMF_DistGridCreate phys decomp', rc)
```
Comment (cacraigucar) [inline src/ionosphere/waccmx/edyn_phys_grid.F90:76]: I realize that this code is most likely a rename of previous code, but I'm seeing a number of return codes which are not being checked here, and below.  Should they be?

**Review by cacraigucar — CHANGES_REQUESTED** [review]

File: src/ionosphere/waccmx/edyn_phys_grid.F90 (hunk 2 above)
Comment (fvitt) [reply src/ionosphere/waccmx/edyn_phys_grid.F90:76]: The return codes from the ESMF routines are checked by `edyn_esmf_chkerr`.  I believe they are all checked.  Am I missing something?

Comment (cacraigucar) [reply src/ionosphere/waccmx/edyn_phys_grid.F90:76]: No - I'm the one missing things (i.e. that edyn_esmf_checkerr is actually doing the checking!)  Sorry about that!

**Review by cacraigucar — APPROVED** [review]

**Pull request merged**